src/west/app/main.py

```diff
@@ -479,6 +491,16 @@ def run_command(self, argv, early_args):
         # If we're running an extension, instantiate it from its
         # spec and re-parse arguments before running.
 
+        # Recursively replace alias command(s) if set
```

Comment:
Please guard this code with something like:
```python
   if not early_args.help and early_args.command_name != "help":
```
(UNTESTED)

This should stop `west help some_alias` and `west -h some_alias` from printing the help of a command resolved 3 or 4 levels down. In the future we want `west help some_alias` to print the alias just like `git help some_alias` does nicely. This guard will help get ready for that. Today it's better for `west help some_alias` to fail or print nothing rather than getting into the bad habit of `west help some_alias` showing the help of the resolved command (XKCD 1172)

Reply:
Hmn, currently all my tests use the `help` command, as it is built-in.